src/main/java/io/github/sskorol/pages/SearchPage.java

```diff
@@ -14,9 +14,9 @@
     private final By inputSearch = By.id("twotabsearchtextbox");
 
     @Step("Search for \"{text}\".")
-    public SearchPage searchFor(final String text) {
+    public ProductPage searchFor(final String text) {
         type(inputSearch, text + Keys.ENTER, enabled);
-        return this;
+        return new ProductPage();
```

Comment:
seems like `this` would be better to use here... check all POs, if there're similar places, which require refactoring...